### PR TITLE
Redirect to login if unauthorized for Deployment View

### PR DESCRIPTION
**Pull request by vsaarinen**
- Adds unauthorized (HTTP status code 401 and 403) handling to API
- Adds support for redirecting to the desired page after logging in
- Store information about whether a request was unauthorized into Errors
- Upgrade some packages
- If a user is unauthorized to open the Deployment View (i.e. `/api/preview/${id}` returns a 401 or 403), they are redirected to the login view and redirected back to the Deployment View once login succeeds. Also support raw deployments.
- If the user is logged in but does not have access rights to the preview, show them an error message instead of redirecting them to the login screen.

Notes:
- This does not add support for redirecting back to all views. In those cases, the user is redirected to the Team Projects View once login succeeds. I feel supporting proper "back-redirects" is enough for the Deployment View. If we want to add proper redirecting support for all views, it requires a larger refactoring that I'd prefer to do in another PR.

TODO:
- [x] Base onto master once #218 is merged.

NOTE: For local development with the backend, minard-ui should be now started with
```
npm start -- http://localtest.me:8000
```

### CODE REVIEW

**Comment by dennari** [comment]
There seems to be an issue with getting the cookie set after logging in in a redirect scenario:

1. Unauthenticated user tries to open the deployment view
2. Gets redirected to the login view
3. If the login is succesful, the deployment view is shown

Apparently the `/team` endpoint, which sets the cookie, doesn't get called in this scenario so access to the raw deployment is blocked.

**Comment by dennari** [comment]
Pretty weird, but after logging in and refreshing I get the login view again.

**Comment by vsaarinen** [comment]
Good catch. The team endpoint was not being called for the Deployment View. Will fix.

File: README.md [hunk 1]
```diff
@@ -53,15 +53,12 @@ yarn start -- <minard-backend API URL> # then open http://localhost:3000/ in a b
 ```
 
 By default, when run locally, the minard-backend (charles) API runs
-on `http://localhost:8000`:
+on port 8000:
 
 ```shell
 yarn start -- http://localhost:8000
```
Comment (dennari) [inline README.md]: Charles's endpoint should now be specified as `http://localtest.me:8000` as otherwise raw deployment authentication won't work

**Comment by vsaarinen** [comment]
I just noticed that opening any view other than the Deployment View as a landing page when you're already logged in doesn't work. Fixing.

**Pull request merged**